### PR TITLE
feat: add workflow vars and search for event types

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Adds New vars for Workflows:
1. "attendee_timezone_info": "Attendee Timezone"
2. "event_start_time_in_attendee_timezone_info": "Event Start Time In Attendee Timezone"
3. "event_end_time_in_attendee_timezone_info": "Event End Time In Attendee Timezone"

Also adds search option in event types, taken mostly from #17088 (@Udit-takkar 👏 )

This pull request introduces several enhancements and refactors across multiple files. The key changes include adding search functionality to the event types listing, updating email and SMS reminders to include attendee timezone information, and refactoring the code for better readability and maintainability.

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

### CODE REVIEW

File: packages/features/ee/workflows/lib/constants.ts [hunk 1]
```diff
@@ -64,6 +64,9 @@ export const DYNAMIC_TEXT_VARIABLES = [
   "reschedule_url",
   "rating_url",
   "no_show_url",
+  "attendee_timezone",
+  "event_time_in_attendee_timezone",
+  "event_end_time_in_attendee_timezone",
 ];
 
 export const FORMATTED_DYNAMIC_TEXT_VARIABLES = ["event_date_", "event_time_", "event_end_time_"];
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/constants.ts:72]: let's add "event_time_in_attendee_timezone" event_end_time_in_attendee_timezone" here too, then date formatting will work for these too

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -2699,6 +2699,9 @@
   "confirm_reassign_unavailable": "Manual Reassign to unavailable host",
   "reassign_unavailable_team_member_description": "Are you sure you want to reassign this booking to an unavailable host?",
   "yes_reassign": "Yes, reassign",
+  "attendee_timezone_info": "Attendee Timezone",
+  "event_time_in_attendee_timezone_info": "Event Start Time In Attendee Timezone",
+  "event_end_time_in_attendee_timezone_info": "Event End Time In Attendee Timezone",
```
Comment (CarinaWolli) [inline apps/web/public/static/locales/en/common.json:2708]: we should add translations for the variable itself too:

attendee_timezone_variable: "Attendee timezone"
event_time_in_attendee_timezone_variable: "Event time in attendee timezone" 
event_end_time_in_attendee_timezone_variable: "Event end time in attendee timezone"

**Review by Udit-takkar — DISMISSED** [review]
One thing that i have to add later on in my PR was to include the search term at code like these apps/web/modules/event-types/views/event-types-listing-view.tsx

```
   const previousValue = utils.viewer.eventTypes.getEventTypesFromGroup.getInfiniteData({
        limit: LIMIT,
        group: { teamId: group?.teamId, parentId: group?.parentId },
      });
```

because for many operation like marking event type as hidden we optimistically mark it as true and then wait for server response.

File: packages/features/ee/workflows/lib/constants.ts [hunk 3]
```diff
@@ -64,6 +64,9 @@ export const DYNAMIC_TEXT_VARIABLES = [
   "reschedule_url",
   "rating_url",
   "no_show_url",
+  "attendee_timezone",
+  "event_time_in_attendee_timezone",
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/constants.ts:68]: should we name it `event_start_time_in_attendee_timezone` instead?

**Review by CarinaWolli — COMMENTED** [review]
still need to test but here's a first review

**Review by zomars — DISMISSED** [review]

**Review by Udit-takkar — COMMENTED** [review]
We have to do the same for 

1) utils.viewer.eventTypes.getEventTypesFromGroup.invalidate in DuplicateDialog
2) utils.viewer.eventTypes.getEventTypesFromGroup.fetchInfinite in packages/lib/hooks/useCreateEventType.ts

And you are not passing debouncedSearchTerm 

Screenshot 2024-10-26 at 2 25 41 AM

**Review by CarinaWolli — DISMISSED** [review]
Looks good! As discussed, let's fix the design of the search bar in a follow-up 🙏

**Comment by alishaz-polymath** [comment]
Addressed all comments 🙏

**Review by zomars — APPROVED** [review]

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**